### PR TITLE
Bug/33200 ps report processing failures after load test

### PR DESCRIPTION
**Pull request by jon-shipley**
Allow the anomaly report to be missing if it was not produced.

### CODE REVIEW

File: functions-app/report-psychometrician-create-download/service/psychometrician-report.service.js [hunk 1]
```diff
@@ -104,11 +106,17 @@ const psychometricianReportService = {
...
     }
 
-    const arStat = await fs.stat(anomalyReportFilename)
-    this.logger.verbose(`${functionName}: anomaly report size: ${Math.round(arStat.size / 1024 / 1024)} MB`)
+    try {
+      const arStat = await fs.stat(anomalyReportFilename)
+      this.logger.verbose(`${functionName}: anomaly report size: ${Math.round(arStat.size / 1024 / 1024)} MB`)
+      filesToZip.push(anomalyReportFilename)
+    } catch (error) {
+      // Anomaly report may not have been produced if there weren't any anomlies
```
Comment (danwilliams) [inline functions-app/report-psychometrician-create-download/service/psychometrician-report.service.js]: Typo: `anomlies` should be `anomalies` 🙂

Comment (jon-shipley) [reply functions-app/report-psychometrician-create-download/service/psychometrician-report.service.js]: Thanks! now fixed.

File: functions-app/report-psychometrician-create-download/service/psychometrician-report.service.js [hunk 2]
```diff
@@ -104,11 +106,17 @@ const psychometricianReportService = {
...
+    try {
+      const arStat = await fs.stat(anomalyReportFilename)
+      this.logger.verbose(`${functionName}: anomaly report size: ${Math.round(arStat.size / 1024 / 1024)} MB`)
+      filesToZip.push(anomalyReportFilename)
+    } catch (error) {
+      // Anomaly report may not have been produced if there weren't any anomalies
+      this.logger.info(`${functionName}: failed to stat ${anomalyReportFilename}: ${error.message}`)
+    }
 
     const zipfileName = `pupil-check-data-${moment().format('YYYY-MM-DD HHmm')}.zip`
```
Comment (GeorgeChatzigiannis) [inline functions-app/report-psychometrician-create-download/service/psychometrician-report.service.js]: @jon-shipley non-blocker: shouldn't that be `zipFilename` ?

Comment (jon-shipley) [reply functions-app/report-psychometrician-create-download/service/psychometrician-report.service.js]: Yes, good spot. 👏 Now fixed.

**Pull request merged**